.github/workflows/test-go.yml

```diff
@@ -269,57 +389,68 @@ jobs:
           # CI commands, like the UI tests, shouldn't have to worry about licensing.
           # So we provide the tests which want an externally supplied license with licenses
           # via the VAULT_LICENSE_CI and VAULT_LICENSE_2 environment variables, and here we unset it.
-          # shellcheck disable=SC2034
-          VAULT_LICENSE=
+          unset VAULT_LICENSE
 
           # Assign test licenses to relevant variables if they aren't already
           if [[ ${{ github.repository }} == 'hashicorp/vault' ]]; then
             export VAULT_LICENSE_CI=${{ secrets.ci_license }}
             export VAULT_LICENSE_2=${{ secrets.ci_license_2 }}
-            export HCP_API_ADDRESS=${{ secrets.HCP_API_ADDRESS }}
-            export HCP_AUTH_URL=${{ secrets.HCP_AUTH_URL }}
-            export HCP_CLIENT_ID=${{ secrets.HCP_CLIENT_ID }}
-            export HCP_CLIENT_SECRET=${{ secrets.HCP_CLIENT_SECRET }}
-            export HCP_RESOURCE_ID=${{ secrets.HCP_RESOURCE_ID }}
-            # Temporarily removing this variable to cause HCP Link tests
-            # to be skipped.
-            #export HCP_SCADA_ADDRESS=${{ secrets.HCP_SCADA_ADDRESS }}
           fi
 
+          # The docker/binary tests are more expensive, and we've had problems with timeouts when running at full
+          # parallelism.  The default if -p isn't specified is to use NumCPUs, which seems fine for regular tests.
+          package_parallelism=""
+
           if [ -f bin/vault ]; then
             VAULT_BINARY="$(pwd)/bin/vault"
             export VAULT_BINARY
+
+            package_parallelism="-p 2"
           fi
 
           # On a release branch, add a flag to rerun failed tests
           # shellcheck disable=SC2193 # can get false positive for this comparision
           if [[  "${{ github.base_ref }}" == release/* ]] || [[  -z "${{ github.base_ref }}" && "${{ github.ref_name }}" == release/* ]]
           then
-           RERUN_FAILS="--rerun-fails"
+            # TODO remove this extra condition once 1.15 is about to released GA
```

Comment:
This TODO needs to be fixed upstream. We'll remove it here so we don't have to backport another fix.